server/api/src/rsDataObjUnlink.cpp

```diff
@@ -291,11 +300,27 @@ int rsMvDataObjToTrash(
 
             auto replica = *id::find_replica(op, std::stoi(cond_input.at(REPL_NUM_KW).value().data()));
 
+            if (const auto ret = ill::try_lock(*op.get(), ill::lock_type::write, replica.replica_number()); ret < 0) {
+                irods::log(LOG_NOTICE, fmt::format(
+                    "[{}:{}] - unlink not allowed because data object is locked",
```

Comment:
do we want logical path / replica number here, too?

Reply:
Done